### PR TITLE
Travis CI Scripts

### PR DESCRIPTION
**Pull request by stephengroat**
Script so that travis can be enabled for the repo

https://travis-ci.org/stephengroat/SVF/builds/234668158

### CODE REVIEW

**Comment by rockysui** [comment]
There is a linking error on my mac (due to the change to CMakeLists.txt), please see below:

Undefined symbols for architecture x86_64:
  "_Cudd_ReadNodeCount", referenced from:
      BddCondManager::getCondNumber() in SrcSnkDDA.cpp.o
  "_Cudd_ReadPeakLiveNodeCount", referenced from:
      BddCondManager::getMaxLiveCondNumber() in SrcSnkDDA.cpp.o
  "_Cudd_bddIthVar", referenced from:
      BddCondManager::createNewCond(unsigned int) in PathCondAllocator.cpp.o
  "_Cudd_Quit", referenced from:
      BddCondManager::~BddCondManager() in PathCondAllocator.cpp.o
  "_Cudd_ReadMemoryInUse", referenced from:
      BddCondManager::getBDDMemUsage() in SrcSnkDDA.cpp.o
  "_Cudd_bddOrLimit", referenced from:
      BddCondManager::OR(DdNode*, DdNode*) in Conditions.cpp.o
  "_Cudd_ReadLogicZero", referenced from:
      BddCondManager::BddZero() const in Conditions.cpp.o
      BddCondManager::BddZero() const in PathCondAllocator.cpp.o
      BddCondManager::BddZero() const in ProgSlice.cpp.o
      BddCondManager::BddZero() const in SrcSnkDDA.cpp.o
  "_Cudd_Ref", referenced from:
      BddCondManager::AND(DdNode*, DdNode*) in Conditions.cpp.o
      BddCondManager::OR(DdNode*, DdNode*) in Conditions.cpp.o
  "_Cudd_ReadOne", referenced from:
      BddCondManager::BddOne() const in Conditions.cpp.o
      BddCondManager::BddOne() const in PathCondAllocator.cpp.o
      BddCondManager::BddOne() const in MemSSA.cpp.o
      BddCondManager::BddOne() const in ProgSlice.cpp.o
  "_Cudd_Init", referenced from:
      BddCondManager::BddCondManager() in PathCondAllocator.cpp.o
      BddCondManager::BddCondManager() in MemSSA.cpp.o
      BddCondManager::BddCondManager() in DoubleFreeChecker.cpp.o
      BddCondManager::BddCondManager() in FileChecker.cpp.o
      BddCondManager::BddCondManager() in LeakChecker.cpp.o
      BddCondManager::BddCondManager() in SrcSnkDDA.cpp.o
  "_Cudd_bddAndLimit", referenced from:
      BddCondManager::AND(DdNode*, DdNode*) in Conditions.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


Could you please also write a brief instruction on how to use your script at the top of the .travis.yml?

Thanks!

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -51,18 +51,10 @@ set(SOURCES
 add_llvm_loadable_module(Svf ${SOURCES})
 add_llvm_Library(LLVMSvf ${SOURCES})
 
-if ( CMAKE_SYSTEM_NAME MATCHES "Darwin")
```
Comment (rockysui) [inline lib/CMakeLists.txt]: I believe the previous one works, your change causes a linking error on my mac.

**Comment by rockysui** [comment]
The CMake works now on both mac and ubuntu.

Any instructions on how to use your travis.yml would be appreciated!

**Comment by stephengroat** [comment]
I'm still getting travis os x errors. i think it's a cmake thing (favoring clang over clang++, but I'm not 100% sure)

do you see this on a clean build? https://travis-ci.org/stephengroat/SVF/jobs/234887344#L1351

**Comment by stephengroat** [comment]
looks good now

File: lib/CUDD/CMakeLists.txt [hunk 2]
```diff
@@ -1,20 +1,20 @@
 set(SOURCES
        cuddAddNeg.c
        cuddBddCorr.c
-       cuddDecomp.c
+       cuddDecomp.cpp
```
Comment (rockysui) [inline lib/CUDD/CMakeLists.txt]: Why did you change the file extensions in CMakeLists.txt? It caused a cmake build crash on my mac.

However, the actual file names in CUDD folder remain unchanged?

Comment (rockysui) [inline lib/CUDD/CMakeLists.txt]: The build is completely ok on my ubuntu machine, but not mac.

Comment (stephengroat) [inline lib/CUDD/CMakeLists.txt]: could you show the error that you're getting? the travis mac builds (and my mac builds) are showing clean.

are you running a clean build (a fresh git clone)?

Comment (rockysui) [inline lib/CUDD/CMakeLists.txt]: See the error file below:
[err.txt](https://github.com/unsw-corg/SVF/files/1021181/err.txt).

Please keep *.c extension instead of "*.cpp" for the third-party lib CUDD. Any reason behind for the changes?

Comment (stephengroat) [inline lib/CUDD/CMakeLists.txt]: could you remove the cmake folder and re-run? that's the issue

i renamed the `c` files to `cpp` because they reference `std11` elements

Comment (rockysui) [inline lib/CUDD/CMakeLists.txt]: CUDD-2.5.0 is a pretty old lib. I believe almost all the CUDD files were written before 2010 (std11). For example, [dddmpStoreAdd.c](https://github.com/unsw-corg/SVF/blob/master/lib/CUDD/dddmpStoreAdd.c) you changed to `cpp` file was written in 2004. There is no std11 feature there:)

In order to make all the CUDD files consistent, could you please leave CUDD file extensions unchanged?

Thanks

**Comment by rockysui** [comment]
Stephen,

Travis looks impressive! Haven't used it before.

Thanks for the contribution!

**Comment by rockysui** [comment]
@stephengroat 

Hi Stephen,

I have made slight changes to the SVF/CMakelist.txt (see https://github.com/unsw-corg/SVF/commit/ed775754ffc24120273677fd699213d03b77491f)


Do you know why Travis build failed under osx configuration (see https://travis-ci.org/unsw-corg/SVF/builds/250650745?utm_source=github_status&utm_medium=notification)?

Thanks

**Comment by rockysui** [comment]
BTW: the CMakeList.txt works well on both my mac and ubuntu machines, but not on Travis servers.

**Comment by stephengroat** [comment]
@rockysui see #25

**Comment by rockysui** [comment]
Thanks!

**Comment by yuleisui** [comment]
Hi Stephen,

Could you please take a look at the Travis error below:

https://travis-ci.org/SVF-tools/SVF?utm_source=email&utm_medium=notification

SVF compiles well on my local machine both on ubuntu and Mac OS

**Comment by yuleisui** [comment]
@stephengroat 


Hi Stephen,

Could you please take a look at the Travis error below:

https://travis-ci.org/SVF-tools/SVF?utm_source=email&utm_medium=notification

SVF compiles well on my local machine both on ubuntu and Mac OS

**Comment by stephengroat** [comment]
@yuleisui homebrew updated `llvm` to 5, `clang-4.0` is no longer available via homebrew

**Comment by yuleisui** [comment]
@stephengroat 
It keeps reminding me the error once after each commit :(

Any idea to fix it, or we may remove the travis CI for mac build?

Thanks
